### PR TITLE
Refine handling of `underperforming_group` issue type

### LINKED ISSUE #1065: lab.find_issues(features=features) outputs error for underperforming issue
lab.find_issues(features=features) output
```
[/Users/sanjana/cleanlab_home/fork_cleanlab/cleanlab/datalab/internal/issue_finder.py:457](https://file+.vscode-resource.vscode-cdn.net/Users/sanjana/cleanlab_home/fork_cleanlab/cleanlab/datalab/internal/issue_finder.py:457): UserWarning: No labels were provided. The 'label' issue type will not be run.
  warnings.warn("No labels were provided. " "The 'label' issue type will not be run.")
Finding null issues ...
Finding outlier issues ...
Fitting OOD estimator based on provided features ...
Finding near_duplicate issues ...
Finding non_iid issues ...
Finding underperforming_group issues ...
Error in underperforming_group: UnderperformingGroupIssueManager.find_issues() missing 1 required positional argument: 'pred_probs'
Failed to check for these issue types: [UnderperformingGroupIssueManager]

Audit complete. 984 issues found in the dataset.
```
Dataset: https://www.kaggle.com/datasets/laotse/credit-risk-dataset/data

Code
```python
import pandas as pd
from cleanlab import Datalab
from sklearn.preprocessing import StandardScaler
import numpy as np

df = pd.read_csv("./credit_risk_dataset.csv")
df = df[~df.isnull().any(axis=1)].copy()
feature_columns = df.columns.to_list()
feature_columns.remove("loan_status")

X_raw = df[feature_columns]
labels = df["loan_status"]

cat_features = [
    "person_home_ownership",
    "loan_intent",
    "loan_grade",
    "cb_person_default_on_file",
]
numeric_features = [
    "person_age",
    "person_income",
    "person_emp_length",
    "loan_amnt",
    "loan_int_rate",
    "loan_percent_income",
    "cb_person_cred_hist_length",
]

X_encoded = pd.get_dummies(X_raw, columns=cat_features, drop_first=True, dtype='float')

scaler = StandardScaler()
X_processed = X_encoded.copy()
X_processed[numeric_features] = scaler.fit_transform(X_encoded[numeric_features])

lab = Datalab({"X": X_processed.to_numpy(), "y": labels})

lab.find_issues(features=X_processed.to_numpy())
```

### PR DESCRIPTION
**Pull request by gogetron**
## Summary
Closes #1065

🎯 Purpose: Refine handling of `underperforming_group` issue type for classification tasks to ensure it is correctly included only when appropriate inputs are provided.

`UnderperformingGroupIssueManager` always requires `pred_probs`. When `features` are available but `pred_probs` is not, the `_CLASSIFICATION_ARGS_DICT` includes the underperforming_group as an available issue type. An additional check was added to remove this issue from the list like we do for `label`, `class_imbalance` and `outlier`. A new assertion was added to the existing test suite. An error is correctly raised in master and this PR fixes the issue.

The `underperforming_group` issue type should only be available when `pred_probs` and another required input (`features`, `knn_graph`, or `cluster_ids`) are provided. This enhancement ensures that the `_CLASSIFICATION_ARGS_DICT` accurately reflects this requirement. An additional check was added to exclude `underperforming_group` from the list of available issue types if these conditions are not met. This refinement aligns with how other issue types such as `label`, `class_imbalance`, and `outlier` are handled. A new assertion was added to the existing test suite to validate this behavior.

## Testing

🔍 Testing Done: Included a new assertion to verify the correct behavior of `underperforming_group` issue type availability based on input conditions.

## Links to Relevant Issues or Conversations

> 🔗 What Git or Slack items (Issues, threads, etc) that are specifically related to
> this work? Please link them here.

- Addresses #1065
- Related to #995
  - A comment was added to `_CLASSIFICATION_ARGS_DICT` to clarify how the mapping should work for `underperforming_group`.

## Reviewer Notes
- The new check for `underperforming_group` issue type and its integration with the `_CLASSIFICATION_ARGS_DICT`.
- The comprehensiveness and coverage of the updated test cases.

### CODE REVIEW

**Review by sanjanag — APPROVED** [review]

**Review by elisno — APPROVED** [review]
LGTM! Made slight improvements to handle the omission of `underperforming_group` more accurately. The additional checks and updated test cases ensure that the issue type is only included when the required inputs are provided.

Thanks @gogetron!
